director/schedule_push.go

```diff
@@ -0,0 +1,254 @@
+package director
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/mdmdirector/mdmdirector/db"
+	"github.com/mdmdirector/mdmdirector/types"
+	"github.com/mdmdirector/mdmdirector/utils"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/vmihailenco/taskq/v3"
+)
+
+func ScheduledCheckin(pushQueue taskq.Queue) {
+
+	var task = taskq.RegisterTask(&taskq.TaskOptions{
+		Name: "push",
+		Handler: func(device types.Device) error {
+			err := PushDevice(device)
+			if err != nil {
+				ErrorLogger(LogHolder{Message: err.Error()})
+			}
+			return nil
+		},
+	})
+
+	counter := 0
+	for {
+		if !DevicesFetchedFromMDM {
+			time.Sleep(30 * time.Second)
+			log.Info("Devices are still being fetched from MicroMDM")
+			counter++
+			if counter > 10 {
+				break
+			}
+		} else {
+			break
+		}
+	}
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+	fn := func() {
+		log.Info("Running scheduled checkin")
+		err := processScheduledCheckin(pushQueue, task)
+		if err != nil {
+			ErrorLogger(LogHolder{Message: err.Error()})
+		}
+	}
+
+	fn()
+
+	for range ticker.C {
+		go fn()
```

Comment:
Any reason to not do this function inline with the go routine?

Reply:
Just because I think this is cleaner as it’s a pretty long function.